src/main/java/lotto/domain/model/result/WinningRank.java

```diff
@@ -0,0 +1,52 @@
+package lotto.domain.model.result;
+
+import java.util.Arrays;
+import java.util.function.BiFunction;
+
+public enum WinningRank {
+
+    NO_WIN(0, false, 0, (count, bonus) -> count <= 0),
+    FIFTH(3, false, 5_000, (count, bonus) -> count == 3),
+    FOURTH(4, false, 50_000, (count, bonus) -> count == 4),
+    THIRD(5, false, 1_500_000, (count, bonus) -> count == 5 && !bonus),
+    SECOND(5, true, 30_000_000, (count, bonus) -> count == 5 && bonus),
+    FIRST(6, false, 2_000_000_000, (count, bonus) -> count == 6);
+
+    private final int countOfMatchingNumbers;
+    private final boolean requiresBonus;
+    private final long prize;
+    private final BiFunction<Integer, Boolean, Boolean> conditionExpress;
```

Comment:
진짜 빈 말이 아니고, 이번 과제 리뷰를 통해서 `enum`의 필드 값으로 함수형 인터페이스 사용하신 분들 덕분에 `enum` 사용에 대한 제 시야가 말도 안되게 넓어져서 너무 감사드립니다!
정말 간편하게 객체 스스로 적절한 `enum` 객체를 반환해 줄 수 있겠네요!

추가적으로 리턴하는 함수형 인터페이스 `BiFunction`은 `BiPredicate`로 대체할 수 있으며, 조건 확인이라는 행위와도 조금 더 연관되니 `Predicate`의 사용도 고려해 보시면 좋을 것 같습니다!
```suggestion
    private final BiPredicate<Integer, Boolean> conditionExpress;
```